src/components/common/PlantingSiteSelector.tsx

```diff
@@ -0,0 +1,44 @@
+import { Dropdown } from '@terraware/web-components';
+import { useAppSelector } from 'src/redux/store';
+import { selectPlantingSites } from 'src/redux/features/tracking/trackingSelectors';
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import strings from 'src/strings';
+
+type PlantingSiteSelectorProps = {
+  onChange: (plantingSiteId: number) => void;
+};
+
+export default function PlantingSiteSelector({ onChange }: PlantingSiteSelectorProps): JSX.Element {
+  // assume `requestPlantingSites` thunk has been dispatched by consumer
+  const plantingSites = useAppSelector(selectPlantingSites);
+  const [selectedPlantingSiteId, setSelectedPlantingSiteId] = useState<number | undefined>();
+
+  const options = useMemo(
+    () => plantingSites?.map((site) => ({ label: site.name, value: site.id })) ?? [],
+    [plantingSites]
+  );
+
+  const updateSelection = useCallback(
+    (newValue: any) => {
+      const id = Number(newValue);
+      setSelectedPlantingSiteId(isNaN(id) ? -1 : id);
+      onChange(isNaN(id) ? -1 : id);
+    },
+    [onChange, setSelectedPlantingSiteId]
```

Comment:
do we need the setSelectedPlantingSiteId?

Reply:
I thought so—I'll try deleting and see if the linter complains...

Reply:
Nope! not needed

Reply:
Thanks for checking \o/